### PR TITLE
(feat): fix query docs and improve offline query

### PR DESCRIPTION
**Pull request by sjmignot**

### CODE REVIEW

File: resolver-online-offline.mdx [hunk 1]
```diff
@@ -30,7 +28,16 @@ including BigQuery, Snowflake, and Redshift.
 In addition, offline queries write their output to a [parquet file](https://parquet.apache.org/)
 in cloud storage (S3/GCS), whereas online queries write their results to database.
 
-[//]: # 'More detail [Chalk architecture](/docs/architecture)'
+| writes to the offline store                                            | writes to the online store                                                          |
+|------------------------------------------------------------------------|-------------------------------------------------------------------------------------|
+| an online query writes all features not read from the online store     | online query which computes features with `max_staleness != 0`                      |
```
Comment (sjmignot) [inline resolver-online-offline.mdx:33]:
```suggestion
| an online query writes all features not read from the online store     | an online query which computes features with `max_staleness != 0`                      |
```

File: resolver-online-offline.mdx [hunk 2]
```diff
@@ -30,7 +28,16 @@ including BigQuery, Snowflake, and Redshift.
 In addition, offline queries write their output to a [parquet file](https://parquet.apache.org/)
 in cloud storage (S3/GCS), whereas online queries write their results to database.
 
-[//]: # 'More detail [Chalk architecture](/docs/architecture)'
+| writes to the offline store                                            | writes to the online store                                                          |
+|------------------------------------------------------------------------|-------------------------------------------------------------------------------------|
+| an online query writes all features not read from the online store     | an online query which computes features with `max_staleness != 0`                      |
```
Comment (sjmignot) [inline resolver-online-offline.mdx:33]:
```suggestion
| an online query writes all freshly computed features (those not read from the online store)     | an online query which computes features with `max_staleness != 0`                      |
```

File: resolver-online-offline.mdx [hunk 3]
```diff
@@ -30,7 +28,16 @@ including BigQuery, Snowflake, and Redshift.
 In addition, offline queries write their output to a [parquet file](https://parquet.apache.org/)
 in cloud storage (S3/GCS), whereas online queries write their results to database.
 
-[//]: # 'More detail [Chalk architecture](/docs/architecture)'
+| writes to the offline store                                            | writes to the online store                                                          |
+|------------------------------------------------------------------------|-------------------------------------------------------------------------------------|
+| an online query writes all freshly computed features (those not read from the online store)     | an online query writes all freshly computed features with `max_staleness != 0`                      |
+| A triggered resolver runs with `store_offline=True` (default behavior) | A triggered resolver runs with `store_online=True` (default behavior)               |
```
Comment (sjmignot) [inline resolver-online-offline.mdx:34]:
```suggestion
| A triggered resolver run with `store_offline=True` (default behavior) | A triggered resolver runs with `store_online=True` (default behavior)               |
```

File: resolver-online-offline.mdx [hunk 4]
```diff
@@ -30,7 +28,16 @@ including BigQuery, Snowflake, and Redshift.
 In addition, offline queries write their output to a [parquet file](https://parquet.apache.org/)
 in cloud storage (S3/GCS), whereas online queries write their results to database.
 
-[//]: # 'More detail [Chalk architecture](/docs/architecture)'
+| writes to the offline store                                            | writes to the online store                                                          |
+|------------------------------------------------------------------------|-------------------------------------------------------------------------------------|
+| an online query writes all freshly computed features (those not read from the online store)     | an online query writes all freshly computed features with `max_staleness != 0`                      |
+| A triggered resolver run with `store_offline=True` (default behavior) | A triggered resolver runs with `store_online=True` (default behavior)               |
```
Comment (sjmignot) [inline resolver-online-offline.mdx:34]:
```suggestion
| A triggered resolver run with `store_offline=True` (default behavior) | A triggered resolver run with `store_online=True` (default behavior)               |
```

File: resolver-online-offline.mdx [hunk 5]
```diff
@@ -30,7 +28,16 @@ including BigQuery, Snowflake, and Redshift.
 In addition, offline queries write their output to a [parquet file](https://parquet.apache.org/)
 in cloud storage (S3/GCS), whereas online queries write their results to database.
 
-[//]: # 'More detail [Chalk architecture](/docs/architecture)'
+| writes to the offline store                                            | writes to the online store                                                          |
+|------------------------------------------------------------------------|-------------------------------------------------------------------------------------|
+| an online query writes all freshly computed features (those not read from the online store)     | an online query writes all freshly computed features with `max_staleness != 0`                      |
+| A triggered resolver run with `store_offline=True` (default behavior) | A triggered resolver run with `store_online=True` (default behavior)               |
+|                                                                        |An offline query with `recompute_features=True` that calculates a feature set with `etl_offline_to_online=True` |
```
Comment (sjmignot) [inline resolver-online-offline.mdx:35]:
```suggestion
|                                                                        |An offline query with `recompute_features=True` that calculates a feature set with `etl_offline_to_online=True` |
```
```suggestion
|                                                                        | An offline query with `recompute_features=True` that calculates a feature set with `etl_offline_to_online=True` |
```

File: resolver-online-offline.mdx [hunk 6]
```diff
@@ -30,7 +28,16 @@ including BigQuery, Snowflake, and Redshift.
...
 in cloud storage (S3/GCS), whereas online queries write their results to database.
 
-[//]: # 'More detail [Chalk architecture](/docs/architecture)'
+| writes to the offline store                                            | writes to the online store                                                          |
+|------------------------------------------------------------------------|-------------------------------------------------------------------------------------|
+| an online query writes all freshly computed features (those not read from the online store)     | an online query writes all freshly computed features with `max_staleness != 0`                      |
+| A triggered resolver run with `store_offline=True` (default behavior) | A triggered resolver run with `store_online=True` (default behavior)               |
+|                                                                        | An offline query with `recompute_features=True` that calculates a feature set with `etl_offline_to_online=True` |
+| scheduled queries with `recompute_features=True` and `store_offline=True (default behavior) | scheduled query with `store_online != False` (default)         |
```
Comment (sjmignot) [inline resolver-online-offline.mdx:36]:
```suggestion
| scheduled queries with `recompute_features=True` and `store_offline=True` (default behavior) | scheduled query with `store_online != False` (default)         |
```

File: resolver-online-offline.mdx [hunk 7]
```diff
@@ -30,7 +28,16 @@ including BigQuery, Snowflake, and Redshift.
...
 in cloud storage (S3/GCS), whereas online queries write their results to database.
 
-[//]: # 'More detail [Chalk architecture](/docs/architecture)'
+| writes to the offline store                                            | writes to the online store                                                          |
+|------------------------------------------------------------------------|-------------------------------------------------------------------------------------|
+| an online query writes all freshly computed features (those not read from the online store)     | an online query writes all freshly computed features with `max_staleness != 0`                      |
+| A triggered resolver run with `store_offline=True` (default behavior) | A triggered resolver run with `store_online=True` (default behavior)               |
+|                                                                        | An offline query with `recompute_features=True` that calculates a feature set with `etl_offline_to_online=True` |
+| scheduled queries with `recompute_features=True` and `store_offline=True` (default behavior) | scheduled query with `store_online != False` (default)         |
```
Comment (sjmignot) [inline resolver-online-offline.mdx:36]:
```suggestion
| scheduled queries with `recompute_features=True` and `store_offline=True (default behavior)` | scheduled query with `store_online != False` (default)         |
```

File: query-overview.mdx [hunk 8]
```diff
@@ -53,19 +53,26 @@ which constrain the outputs to a specific temporal range.
 
 ## Query Side Effects
 
-Chalk queries return and write data. This is an essential part of
-Chalk: every time you compute a feature, either through an online or offline
-query, the output is written down in the offline store. This makes it easy to:
+Chalk queries can also write data. This is an essential part of
+Chalk: every time you compute a feature through an online query, the
```
Comment (sjmignot) [inline query-overview.mdx:57]:
```suggestion
Chalk: every time you compute a feature through an online query the
```

File: training-client.mdx [hunk 9]
```diff
@@ -4,44 +4,64 @@ description: Fetch offline feature values.
...
+  requested feature.
+</TipInfo>
+
+In an offline query you can request multiple instances of a feature set. Additionally, by default,
+an offline query returns all matches for a primary key in the offline store. For instance, if you
+requested the `num_transactions` feature for users with id's `1`, `2`, or `3`, you could receive
+multiple rows for each user if you've computed the `num_transactions` feature for each user multiple
+times.
+
+The main use cases for offline queries is creating datasets.
```
Comment (AndyMoreland) [inline training-client.mdx:25]: use case

File: training-client.mdx [hunk 10]
```diff
@@ -4,44 +4,64 @@ description: Fetch offline feature values.
...
+
+Chalk supports a number of client that can run offline queries. In this section, our examples focus
```
Comment (AndyMoreland) [inline training-client.mdx:27]: clients

File: training-client.mdx [hunk 11]
```diff
@@ -4,44 +4,64 @@ description: Fetch offline feature values.
...
+from datetime import datetime
+
+client = ChalkClient()
+client.offline_query(
+  input={'user.id': [1,2,3,4]},         # Input
+  output=['user.name'],                 # Output
+  # tags=['test'],                      # Environment
+  # branch='branch',
+  # recompute_features=True,            # Run Resolvers
+  # run_async=True,                     # Run Configuration
```
Comment (AndyMoreland) [inline training-client.mdx:49]: 'run_asynchronously', unfortunately.

File: training-client.mdx [hunk 12]
```diff
@@ -4,44 +4,64 @@ description: Fetch offline feature values.
...
 
 ### Input
 
 As [input](/api-docs#ChalkClient.offline_query.input), `offline_query` takes a
-[chalk.DataFrame](/docs/dataframe)
-or
-[pandas.DataFrame](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html)
+[chalk.DataFrame](/docs/dataframe) or [pandas.DataFrame](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html)
 with one column for each known feature in the input, and one column
 with the heading of `chalk.features.timestamp`.
```
Comment (AndyMoreland) [inline training-client.mdx:66]: I think this is actually incorrect (the chalk.features.timestamp bit)

**Review by AndyMoreland — APPROVED** [review]

**Pull request merged**